lib/async-emitter.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const { iter } = require('@metarhia/common');
+
+class AsyncEmitter {
+  constructor() {
+    this.events = new Map();
+    this.wrappers = new Map();
+  }
+
+  on(name, fn) {
+    let event = this.events.get(name);
+    if (!event) {
+      event = new Set();
+      this.events.set(name, event);
+    }
+    event.add(fn);
+  }
+
+  once(name, fn) {
+    if (fn === undefined) {
+      return new Promise(resolve => {
+        this.once(name, resolve);
+      });
+    }
+    const wrapper = (...args) => {
+      this.remove(name, wrapper);
+      return fn(...args);
+    };
+    this.wrappers.set(fn, wrapper);
```

Comment:
```js
e.once('event', f);
e.once('event', f);
```

In such a case, `f` will be called only once.

Reply:
I think it's ok for us, this isn't a polyfill of `EvenEmitter`, so `on` work in a same way:
```js
e.on('event', f);
e.on('event', f);
```
I don't see real usable cases for double subscription for `on` and `once` @nechaido